src/admin/index.js

```diff
@@ -247,13 +248,35 @@ module.exports = ({
         await cluster.refreshMetadataIfNecessary()
 
         const metadata = cluster.findTopicPartitionMetadata(topic)
+
         const high = await cluster.fetchTopicsOffset([
           {
             topic,
             fromBeginning: false,
             partitions: metadata.map(p => ({ partition: p.partitionId })),
           },
         ])
+        const { partitions: highPartitions } = high.pop()
+
+        if (fromTimestamp != null) {
+          const offsets = await cluster.fetchTopicsOffset([
+            {
+              topic,
+              fromTimestamp,
+              partitions: metadata.map(p => ({ partition: p.partitionId })),
+            },
+          ])
+          const { partitions } = offsets.pop()
+
+          return partitions.map(({ partition, offset }) => ({
```

Comment:
This breaks the type definition for this method, as it's missing `high` and `low`: https://github.com/tulios/kafkajs/blob/de776226f1d7c14b506105c0e9efae257485c984/types/index.d.ts#L317-L319

In the long term, I think we should split this method up into three, `fetchBeginningOffset`,  `fetchEndOffset` and `fetchOffsetByTimestamp`, but obviously that would be a breaking change. In the meantime, maybe we should avoid bloating this method more and just move this out to `fetchOffsetByTimestamp` right away.

The other option would be to figure out what to do with `low` and `high` when a timestamp is passed in. The most obvious solution to me is to have them behave just like today, but it does feel pretty unintuitive when you are passing in a timestamp to get offsets that are outside that timestamp.

Reply:
Agree, I'll move this to a new function `fetchOffsetByTimestamp`.

Reply:
Extracted.